binary_gentoo/internal/cli/build.py

```diff
@@ -176,19 +182,27 @@ def build(config):
         f'LDFLAGS={shlex.quote(config.ldflags)}',
     ]
 
+    if config.tag_docker_image is not None:
+        # TODO there should probably be some sanity checks on the provided image name here
+        container_name = f'binary-gentoo-{uuid.uuid4().hex}'
```

Comment:
I think that comment can be dropped: I don't see us adding more safety checks here, seems perfectly safe already. Am I missing something?

Reply:
It's a comment to check the image name passed by `--tag-docker-image`. If it isn't a valid image name (or worse, some malicous code - don't know if that would be possible) it would create problems after the image has been created, at the moment the `docker commit` is ran. It would be better to catch them upfront.

Reply:
That makes total sense *but* were taking full control of the container name by `container_name = f'binary-gentoo-{uuid.uuid4().hex}'`. Doesn't that mean we're 100% safe?

Reply:
I believe we have a miscommunication here. You are referring to the _container_ name, which is indeed 100% safe because it is generated in the code as you mention. The TODO comment is referring to the _image name_, which is provided by the user on the command line as argument `--tag-docker-image`.

To make it even more visible:
```
(build.py line 339)
announce_and_call(['docker', 'commit', container_name, config.tag_docker_image])
                                       ^^^             ^^^
                        [container name, safe]     [image name, user input]
```

Reply:
You're right, I was mis-reading "image" for "container"; I think the location of the comment tricked me.
We're only forwarding the argument and we're invoking `docker commit` without a layer of shell in between so unless `docker commit` has its own user input interpretation problem, I think we're safe.

```console
# docker commit 07dfbd238644 'hello world'
invalid reference format

# docker commit 07dfbd238644 debian-sid-dummy-tag
sha256:618761729ceb6e20593e3717e03bdcab907f98a69bba75f0bd0c5fee8e46394d
```

My vote for dropping the comment. The alternative would be to do our our own validation with false positives potential on top before a merge, but I don't see much value in that extra code myself at the moment. What do you think?

Reply:
Indeed, since it's only passed along, the security risk is minimal and docker should handle it. Also, I guess we are in a scenario where the person who installs `binary-gentoo`, will be its main user. So it's not like we need to treat this input as hostile, the input given with `--tag-docker-image` should come from a trusted source. I don't mind dropping the comment, just need to remember that if ever the image name is used in Python code, it's not "clean" data but raw user input.

Reply:
> Indeed, since it's only passed along, the security risk is minimal and docker should handle it.

Good.

> Also, I guess we are in a scenario where the person who installs `binary-gentoo`, will be its main user. So it's not like we need to treat this input as hostile, the input given with `--tag-docker-image` should come from a trusted source. I don't mind dropping the comment, just need to remember that if ever the image name is used in Python code, it's not "clean" data but raw user input.

That's not a threat model that I feel super comfortable with. I want the code to be robust in the sense that things that are nor explicitly expecting shell input will never be interpreted by shell. At the same time, marking user input in code everywhere would be rather noisy, and not be special enough to be marked as special, I think. So I'd expect that question to be hanging in the air before changing or extending use of a variable: Is the is user controlled, yes or no. Just like a general security principle to create tempfiles in a safe way rather than a vulnerable one.

Reply:
I think you're saying the same thing. Basically: we remove the comment here, but if ever the value of `--tag-docker-image` is used inside the Python code for something else, then you expect the developer to think whether additional checks are required for security purposes. Is that correct?

In any case, we agree then to remove the comment here.

Reply:
> if ever the value of `--tag-docker-image` is used inside the Python code for something else, then you expect the developer to think whether additional checks are required for security purposes. Is that correct?

Yes, excellent.